src/common/utils/resolve_reference.py

```diff
@@ -128,9 +126,9 @@ class AttributeItem:
     def __repr__(self):
         return f'Attribute="{self.path}"'
 
-    def get_attribute(
-        self, document: dict | list, data_source: DataSource, get_data_source: Callable
-    ) -> tuple[Any, str]:
+    def get_child(self, document: dict | list, data_source: DataSource, get_data_source: Callable) -> tuple[Any, str]:
+        if isinstance(document, dict) and is_reference(document):
```

Comment:
is_reference already checks if document is type, so in theory the extra type check should be unnecessary. However, mypy didn't understand that, so it raised a pre-commit error